### ISSUE TITLE
Default `kbi` weight quantizer silently zeros standard-initialized weights

### ISSUE THREAD

**Issue by LarocheC** [calad0i/HGQ2#11]
## Environment

HGQ 0.1.8, Keras 3 with the torch backend

## Summary

I ran into what looks like a potentially surprising default behavior with the
`kbi` weight quantizer.

With the default weight config (`b0=4, i0=2`), the number of fractional bits is `f = b - i = 2` the quantization step is `2^-2 = 0.25`. This means that weights with `|w| < 0.125` are rounded to zero.

For a standard Glorot/He initialization, at least for reasonably sized layers, many weights are typically around this scale or smaller. In the example below, the whole kernel is quantized to zero immediately after `build()`.

This was a bit hard to spot because there is no warning or error. Training still runs, but the model starts from a degenerate all-zero quantized kernel.

## Reproduction

```python
import numpy as np
import keras.ops as ops
from hgq.layers import QDense

l = QDense(256)
l.build((None, 256))   # Glorot init, as for a regular Keras layer

qk = ops.convert_to_numpy(l.qkernel)
print(f"quant kernel: std={qk.std():.5f}  nonzero={100*np.mean(qk != 0):.1f}%")
# -> quant kernel: std=0.00000  nonzero=0.0%
```

## Impact

In my experiments, this made training difficult to recover from.

The straight-through estimator still passes gradients to the underlying floating point weights, so in principle the model can recover. However, if the quantized kernel of a layer starts entirely at zero, the downstream activations are also zero, and the model seems to need to “wake up”  rogressively from the input side.

At the same time, the bit-width regularizer can keep pushing the precision down during this initial dead phase, which seems to make recovery harder.

## Suggested improvement

A minimal improvement could be to emit a warning when a weight quantizer rounds all, or almost all, of a kernel to zero at build time. This would not require an API change, but would make the issue much easier to diagnose.

## Workaround

The workaround I found is to increase the initial bit-width for weight quantizers around model construction:

```python
from hgq.config import QuantizerConfigScope

# b0=12, i0 stays 2  ->  f = 10  ->  grid step 2^-10 ≈ 0.001
with QuantizerConfigScope(q_type='kbi', place='weight', b0=12):
    model = build_model()
```

With this change, around 99.5% of the initialized weights stay non-zero in my test, compared to 0.0% with the default config.

This seems safe in practice, since `b` is trainable and the bit-width regularizer can still reduce the precision during training. In other words,
`b0` only controls the starting grid, not necessarily the final precision.

**Comment by calad0i** [comment]
Thanks for the issue.

I increased the default to 8 bits (fixed the example) and adds a warning if the mse changed by >50% by quantization. Closing as completed, please re-open if there is any further suggestion/questions.
